src/Modules/Data/Commands.Data.Test/AddPowerBIDatasetTests.cs

```diff
@@ -0,0 +1,109 @@
+
+        [TestMethod]
+        [TestCategory("Interactive")]
+        [TestCategory("SkipWhenLiveUnitTesting")] // Ignore for Live Unit Testing
+        public void EndToEndAddPowerBIDataset()
+        {
+            using (var ps = System.Management.Automation.PowerShell.Create())
+            {
+                // Arrange
+                ProfileTestUtilities.ConnectToPowerBI(ps, PowerBIEnvironmentType.Public);
+                ps.AddCommand(NewPowerBIColumnCmdletInfo).AddParameter(nameof(NewPowerBIColumn.Name), "Col1").AddParameter(nameof(NewPowerBIColumn.DataType), PowerBIDataType.String);
+                var col1 = ps.Invoke();
+                ps.Commands.Clear();
+                ps.AddCommand(NewPowerBIColumnCmdletInfo).AddParameter(nameof(NewPowerBIColumn.Name), "Col2").AddParameter(nameof(NewPowerBIColumn.DataType), PowerBIDataType.Boolean);
+                var col2 = ps.Invoke();
+                ps.Commands.Clear();
+                var columns = new List<Column>() { col1.First().BaseObject as Column, col2.First().BaseObject as Column };
+                ps.AddCommand(NewPowerBITableCmdletInfo).AddParameter(nameof(NewPowerBITable.Name), "Table1").AddParameter(nameof(NewPowerBITable.Columns), columns);
+                var table1 = ps.Invoke();
+                var tables = new List<Table>() { table1.First().BaseObject as Table };
+                ps.Commands.Clear();
+                ps.AddCommand(NewPowerBIDatasetCmdletInfo).AddParameter(nameof(NewPowerBIDataset.Name), "MyDataSet").AddParameter(nameof(NewPowerBIDataset.Tables), tables);
+                var dataset = ps.Invoke();
+                ps.Commands.Clear();
+
+                ps.AddCommand(AddPowerBIDatasetCmdletInfo)
+                    .AddParameter(nameof(AddPowerBIDataset.Dataset), dataset.First().BaseObject as Dataset);
+                
+                // Act
+                var result = ps.Invoke();
+
+                // Assert
+                TestUtilities.AssertNoCmdletErrors(ps);
+                Assert.IsNotNull(result);
+                Assert.IsTrue(result.Count > 0);
+            }
+        }
+
+        [TestMethod]
+        public void AddPowerBIDataset_IndividualScope_DatasetParameterSetName()
+        {
+            var expectedDataset = new Dataset { Id = Guid.NewGuid(), Name = "TestDataset" };
+            var client = new Mock<IPowerBIApiClient>();
+            client.Setup(x => x.Datasets.AddDataset(expectedDataset, null)).Returns(expectedDataset);
+            var initFactory = new TestPowerBICmdletInitFactory(client.Object);
+            var cmdlet = new AddPowerBIDataset(initFactory);
+
+            cmdlet.Dataset = expectedDataset;
+
+            // Act
+            cmdlet.InvokePowerBICmdlet();
+
+            // Assert
+            Assert.AreEqual(expectedDataset.Name, (initFactory.Logger.Output.First() as Dataset).Name);
```

Comment:
I think you should be able to use initFactory.AssertExpectedUnitTestResults if you wrap the expectedDataset object in an array. You could also use TestUtilities.AssertExpectedUnitTestResults instead.

Reply:
I initially tried using it but it takes List<T> rather than T. Do we want to create expectedresults as List?

Reply:
I think taking in a list or array should be fine.

Reply:
my bad, it was my misunderstanding. I didnt need array anyway. I fixed it.